### PR TITLE
Adding NWA3D A5 printer

### PR DESCRIPTION
**Pull request by DragonJe**
I created the definition, extruder, profiles, and uploaded a backplate png for the NWA3D A5 3D printer to be listed in Cura.

Thanks!

### CODE REVIEW

**Comment by fieldOfView** [comment]
This PR includes an unused backplate png. Unless you specify a platform mesh with UV coordinates, don't include the backplate image.

**Comment by nallath** [comment]
For our reference; CURA-6079

**Comment by DragonJe** [comment]
Ok, I don't have a platform mesh, so I just deleted it for now. I might make one and add it later.

Thanks for the help!

File: resources/definitions/nwa3d_a5.def.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+    "name": "NWA3D A5",
+    "version": 1,
```
Comment (diegopradogesto) [inline resources/definitions/nwa3d_a5.def.json]: Currently it has to be `"version": 2`
```suggestion
    "version": 2,
```

File: resources/extruders/nwa3d_a5_extruder_0.def.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "id": "nwa3d_a5_extruder_0",
+    "version": 1,
```
Comment (diegopradogesto) [inline resources/extruders/nwa3d_a5_extruder_0.def.json]: It has to be `"version": 2`
```suggestion
    "version": 2,
```

File: resources/quality/nwa3d_a5/nwa3d_a5_fast.inst.cfg [hunk 3]
```diff
@@ -0,0 +1,124 @@
+[general]
+version = 1
```
Comment (diegopradogesto) [inline resources/quality/nwa3d_a5/nwa3d_a5_fast.inst.cfg]: It has to be `version = 4`
```suggestion
version = 4
```

File: resources/quality/nwa3d_a5/nwa3d_a5_fast.inst.cfg [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+name = Fast Quality
+definition = nwa3d_a5
+
+[metadata]
+setting version = 6
```
Comment (diegopradogesto) [inline resources/quality/nwa3d_a5/nwa3d_a5_fast.inst.cfg]:
```suggestion
setting_version = 6
```

File: resources/quality/nwa3d_a5/nwa3d_a5_normal.inst.cfg [hunk 5]
```diff
@@ -0,0 +1,124 @@
+[general]
+version = 1
```
Comment (diegopradogesto) [inline resources/quality/nwa3d_a5/nwa3d_a5_normal.inst.cfg]:
```suggestion
version = 4
```

File: resources/quality/nwa3d_a5/nwa3d_a5_normal.inst.cfg [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+name = Normal Quality
+definition = nwa3d_a5
+
+[metadata]
+setting version = 6
```
Comment (diegopradogesto) [inline resources/quality/nwa3d_a5/nwa3d_a5_normal.inst.cfg]:
```suggestion
setting_version = 6
```

File: resources/definitions/nwa3d_a5.def.json [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        "has_variants": false,
+        "has_machine_materials": true,
+        "has_variant_materials": false,
+        "preferred_quality_type": "normal",
+        "has_machine_quality": true,
+        "preferred_material": "generic_pla",
+        "machine_extruder_trains": 
+        {
+            "0": "nwa3d_a5_extruder_0"
+        },
```
Comment (diegopradogesto) [inline resources/definitions/nwa3d_a5.def.json]: Remove `,`
```suggestion
        }
```

File: resources/definitions/nwa3d_a5.def.json [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+            "default_value": false 
+        },
+        "material_diameter": {
+            "default_value": 1.75
+        },
+        "machine_gcode_flavor": { 
+            "default_value": "RepRap (RepRap)" 
+        },
+        "machine_start_gcode": { 
+            "G28 ; Home\nG1 Z15.0 F6000 ; Move Z axis up 15mm\n ; Prime the extruder\nG92 E0\nG1 F200 E3\nG92 E0"
```
Comment (diegopradogesto) [inline resources/definitions/nwa3d_a5.def.json]: Missing `default_value`
```suggestion
            "default_value": "G28 ; Home\nG1 Z15.0 F6000 ; Move Z axis up 15mm\n ; Prime the extruder\nG92 E0\nG1 F200 E3\nG92 E0" 
```

File: resources/definitions/nwa3d_a5.def.json [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+            "default_value": 1.75
+        },
+        "machine_gcode_flavor": { 
+            "default_value": "RepRap (RepRap)" 
+        },
+        "machine_start_gcode": { 
+            "G28 ; Home\nG1 Z15.0 F6000 ; Move Z axis up 15mm\n ; Prime the extruder\nG92 E0\nG1 F200 E3\nG92 E0" 
+        },
+        "machine_end_gcode": { 
+            "M104 S0\nM140 S0\n ; Retract the filament\nG92 E1\nG1 E-1 F300\nG28 X0 Y0\nM84"
```
Comment (diegopradogesto) [inline resources/definitions/nwa3d_a5.def.json]: Missing `default_value`
```suggestion
            "default_value": "M104 S0\nM140 S0\n ; Retract the filament\nG92 E1\nG1 E-1 F300\nG28 X0 Y0\nM84" 
```

File: resources/definitions/nwa3d_a5.def.json [hunk 10]
```diff
@@ -0,0 +1,65 @@
+{
+    "name": "NWA3D A5",
+    "version": 1,
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "DragonJe",
+        "manufacturer": "NWA 3D LLC",
+        "file_formats": "text/x-gcode",
+        "platform_texture": "nwa3d_backplate.png",
```
Comment (diegopradogesto) [inline resources/definitions/nwa3d_a5.def.json]: File doesn't exist
```suggestion
```

**Comment by fieldOfView** [comment]
The quality profiles in this PR contain lots of references to non-existing settings. Eg `travel_speed` should have been `speed_travel`, `initial_layer_travel_speed` should have been `speed_travel_layer_0`. It looks like the setting names in the profile were written like the setting labels in the sidebar with the spaces replaced by underscores. @DragonJe, that's not how it works, here is a list of variable names (taken from fdmprinter.def.json):
http://files.fieldofview.com/cura/Replacement_Patterns.html

The quality profiles also overspecify settings (eg specifying all of top_bottom_thickness, top_thickness, top_layers, bottom_thickness and bottom_layers, all as literal values).

Basically, these profiles won't work as expected.

**Comment by DragonJe** [comment]
Thanks @fieldOfView. I'll make a new request with the correct names. That Replacement Patterns page is just what I needed!
